processing/src/main/java/org/apache/druid/segment/incremental/IncrementalIndexAdapter.java

```diff
@@ -212,21 +217,22 @@ public IntIterator iterator()
     }
   }
 
-  @Override
-  public String getMetricType(String metric)
-  {
-    return index.getMetricType(metric);
-  }
-
-  @Override
-  public ColumnCapabilities getCapabilities(String column)
+  private static class DimensionAccessor
   {
-    return index.getColumnCapabilities(column);
-  }
+    private final IncrementalIndex.DimensionDesc dimensionDesc;
+    @Nullable
+    private final MutableBitmap[] invertedIndexes;
+    private final DimensionIndexer indexer;
 
-  @Override
-  public Metadata getMetadata()
-  {
-    return index.getMetadata();
+    public DimensionAccessor(IncrementalIndex.DimensionDesc dimensionDesc)
+    {
+      this.dimensionDesc = dimensionDesc;
+      this.indexer = dimensionDesc.getIndexer();
+      if (dimensionDesc.getCapabilities().hasBitmapIndexes()) {
+        this.invertedIndexes = new MutableBitmap[indexer.getCardinality() + 1];
+      } else {
+        this.invertedIndexes = null;
+      }
+    }
```

Comment:
It took me a bit to realize this, but the changes in this class are all whitespace right?  I.e. code lines were moved, but nothing actually changed?

Reply:
ah yeah, i moved some stuff around, `getIncrementalIndex` is a newly added method though